opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingDynamicFeature.java

```diff
@@ -21,16 +21,19 @@
  */
 @Provider
 public class ServerTracingDynamicFeature implements DynamicFeature {
-
     private static final Logger log = Logger.getLogger(ServerTracingDynamicFeature.class.getName());
 
     private Builder builder;
 
+    /**
+     * When using this constructor application has to call {@link GlobalTracer#register} to register
+     * tracer instance. Ideally it should be called in {@link javax.servlet.ServletContextListener}.
+     */
     public ServerTracingDynamicFeature() {
         this(new Builder(GlobalTracer.get()));
     }
 
-    private ServerTracingDynamicFeature(Builder builder) {
+    public ServerTracingDynamicFeature(Builder builder) {
```

Comment:
You are designing an API. Every move you make should be well thought of: if you restrict access, you should know why. If you give access, you should know why :)

If you don't have a reason to give or restrict access, better to restrict now and give later, as it's harder to restrict it later if it's open today.